### PR TITLE
Allow more flexibility in defining where the coach mark content shoul…

### PR DESCRIPTION
**Pull request by ojj11**
…d appear in the punch hole coach mark

Sorry to pester you both again - but as you've already reviewed this I thought it would be easier for you to get up to speed on this new commit @beatrizz @lachiemurray 

I fixed up the test to make sure it'll break when the coach mark content is shown over the top of the punch hole, and fixed it so this now passes. I also added a builder method to explicitly define what the layout parameters should be, otherwise defaulting to what they were previously

### CODE REVIEW

File: cornedbeef/src/main/java/com/swiftkey/cornedbeef/PunchHoleCoachMark.java [hunk 1]
```diff
@@ -7,28 +7,56 @@
 import android.content.Context;
 import android.content.res.Configuration;
 import android.os.Build;
+import android.support.annotation.IntDef;
 import android.support.annotation.LayoutRes;
+import android.util.Log;
```
Comment (lachiemurray) [inline cornedbeef/src/main/java/com/swiftkey/cornedbeef/PunchHoleCoachMark.java]: Unused import?

**Comment by beatrizz** [comment]
Could you please add screenshots to show what the new positions look like?

File: cornedbeef/src/main/java/com/swiftkey/cornedbeef/PunchHoleCoachMark.java [hunk 2]
```diff
@@ -235,6 +284,26 @@ public PunchHoleCoachMarkBuilder setOverlayColor(int overlayColor) {
...
+         * @param contentHeight the height of the content - defaults to {@link WRAP_CONTENT}
+         * @param contentPositioning where to place the content - defaults to
+         *  {@link POSITION_CONTENT_AUTOMATICALLY}
+         */
+        public PunchHoleCoachMarkBuilder setContentLayoutParams(
+                final int contentWidth,
+                final int contentHeight,
+                @PunchMarkContentPosition final int contentPositioning) {
+
+            this.contentWidth = contentWidth;
```
Comment (beatrizz) [inline cornedbeef/src/main/java/com/swiftkey/cornedbeef/PunchHoleCoachMark.java]: Why do we need the content width and height? If we pass the view as a parameter, can't it have the width and height already set?

Comment (ojj11) [inline cornedbeef/src/main/java/com/swiftkey/cornedbeef/PunchHoleCoachMark.java]: I don't think that's a nice API for the integrator - after all what type of LayoutParams should coach mark content use?

**Comment by ojj11** [comment]
@beatrizz tagged you on the internal ticket that has screenshots

**Comment by beatrizz** [comment]
What happens if we want the content above or below explicitly but there's not enough space?

**Comment by ojj11** [comment]
@beatrizz I guess it'll just be all cramped up, or will overflow across the punch hole, but I don't think we can have an API that makes it explicitly above or below the punch hole, whilst also overriding that in some instances...

**Comment by beatrizz** [comment]
Yep, I also don't know what happened before if there was no space.

:ship: :it:

**Comment by lachiemurray** [comment]
:shipit:

**Pull request merged**